### ISSUE TITLE
`prefer-node-remove`: Allow more types of `childNode`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#549]
Ref: https://github.com/sindresorhus/eslint-plugin-unicorn/pull/507#issuecomment-585156112

Currently, this rule only check that `childNode` is `ThisExpression` or `Identifier`, should allow more AST Types, like:

```js
parentNode.removeChild(some.node);
                       ^^^^^^^^^ `MemberExpression`
parentNode.removeChild(getChildNode());
                       ^^^^^^^^^^^^^^ `CallExpression`
```

More cases: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/fe711c851b75d663c04bfa1c2e07649cb7ba58c0/test/prefer-node-remove.js#L61-L65

Note: some case may hard to fix.

Eg:

```js
parentNode.removeChild(await getChild());

// ->
await getChild().remove(); // Not right

// Should be parenthesed
(await getChild()).remove();

// Even might need semicolon
const foo = []
;(await getChild()).remove();

```

Another related issue: https://github.com/sindresorhus/eslint-plugin-unicorn/issues/548
